### PR TITLE
Markdown api definition with use of core-component

### PR DESCRIPTION
**Pull request by mryave**
## Hey, I just made a Pull Request!
adds option to present api documentation in markdown format. Example here is for grpc documentation generated by [protoc-gen-doc](https://github.com/pseudomuto/protoc-gen-doc) . 

Open question: Should we extend the use of `MarkdownContent` in @backstage/core-components to allow the option of rendering trusted html as offered by the component it uses under the hood (ie `ReactMarkdown`)? 
Or, should the feature of rendering trusted html content be scoped specifically to the markdown api documentation implementation?

example use of core-component extension in https://github.com/backstage/backstage/pull/10973 .
Apr-21-2022 15-33-49

#### :heavy_check_mark: Checklist

- [x] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [x] Added or updated documentation
- [x] Tests for new functionality and regression tests for bug fixes
- [x] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

**Comment by mryave** [comment]
@Rugvip @jrusso1020 This is the implementation for the markdown api component. Would it be more fitting for me to not add to the existing @backstage/core-components `MarkdownContent` component and instead take a different route? 

The implementation would be refactored to do the same thing, allow markdown and trusted html content to be rendered, but specifically for the api documentation rather than allowing anyone to make that decision anywhere while using @backstage/core-component's `MarkdownContent`

File: packages/core-components/src/components/MarkdownContent/MarkdownContent.test.tsx [hunk 1]
```diff
@@ -62,4 +62,20 @@ describe('<MarkdownContent />', () => {
     expect(fp2).toBeInTheDocument();
     expect(rendered.getByText(');', { selector: 'span' })).toBeInTheDocument();
   });
+  it('Render MarkdownContent component with trusted HTML content', async () => {
+    const rendered = await renderWithEffects(
+      wrapInTestApp(
+        <MarkdownContent
+          content="```<blockquote> May not look like much but got it where it counts</blockquote>```"
```
Comment (freben) [inline packages/core-components/src/components/MarkdownContent/MarkdownContent.test.tsx:69]: hm isn't this double blockquotes basically? should you remove the triple backticks?

File: plugins/api-docs/src/components/MarkdownApiDefinitionWidget/MarkdownApiDefinitionWidget.tsx [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import { MarkdownContent } from '@backstage/core-components';
+
+export type MarkdownApiDefinitionWidgetProps = {
+  definition: any;
```
Comment (freben) [inline plugins/api-docs/src/components/MarkdownApiDefinitionWidget/MarkdownApiDefinitionWidget.tsx:21]: isn't this meant to be a string?

**Review by freben — COMMENTED** [review]
Looks OK to me. @Rugvip any additional concerns? There are two smaller comments herein too

**Review by Rugvip — COMMENTED** [review]
I would like to make sure that https://github.com/backstage/backstage/pull/10973#issuecomment-1112081902 is addressed, since those changes are included here. More specifically making sure that either the entire `MarkdownContent` component is lazy loaded, or that we lazy load the `rehype-raw` plugin.

**Comment by mryave** [comment]
Hey all! circling back to this. will have a look and update this PR. thank you for getting back to me on this!

**Pull request closed without merging**